spicy/runtime/include/util.h

```diff
@@ -31,4 +33,36 @@ extern const hilti::rt::Vector<
     std::optional<std::tuple<hilti::rt::integer::safe<uint64_t>, std::optional<hilti::rt::integer::safe<uint64_t>>>>>*
 get_offsets_for_unit(const hilti::rt::type_info::Struct& struct_, const hilti::rt::type_info::Value& value);
 
+/** Confirm a unit in trial mode. */
+template<typename U>
+inline void confirm(U& p) {
+    // If we are not in trial mode `confirm` is a no-op.
+    if ( p.__trial_mode ) {
+        p.__trial_mode.reset();
+
+        // TODO(bbannier): For consistence we would ideally bracket the hook
+        // invocation with calls to `ParserBuilder::beforeHook` and
+        // `afterHook`, but this is not possible since we have no direct access
+        // to the parser state here.
```

Comment:
As discussed offline, the current behavior can be surprising and not what users want, e.g.,

```.ruby
module foo;

type X = unit {
	a: uint8;
};

public type Y = unit {
	x: X;

	on %done {
		self.x.confirm(); # Has no effect since `self.x` is not the current unit.
	}
};
```

Let's instead introduce Spicy keywords `confirm` and `reject` which do the expected thing (and should probably trigger at least a codegen error if invoked in contexts w/o parser state).